app/src/org/commcare/views/widgets/CalendarFragment.java

```diff
@@ -164,9 +164,15 @@ public void onNothingSelected(AdapterView<?> parent) {}
         });
     }
 
+    private int getMaxYearForCalendar() {

```

Comment:
Fix number 1: the max year for the year dropdown in the calendar fragment should be set based on the actual current year (like the current phone's date), not the currently-selected year